### PR TITLE
Fixed an issue with user management in portal groups

### PR DESCRIPTION
**Pull request by valadas**
- Fixed an issue with user management in portal groups
- Fixed many of the build warnings, there are a couple left about IPortalSettings vs PortalSettings which I did not fix because I think the correct solution would involve modifying other components so they can take IPortalSettings instead of PortalSettings and well, lack of time on that...

### CODE REVIEW

File: DNN Platform/Library/Entities/Users/UserController.cs [hunk 1]
```diff
@@ -649,35 +647,34 @@ public static string GetPassword(ref UserInfo user, string passwordAnswer)
             return user.Membership.Password;
         }
 
+        /// <summary>
+        /// Gets the unauthorized users for a portal.
+        /// </summary>
+        /// <param name="portalId">The site (portal) id on which to search for Unauthorized users.</param>
+        /// <param name="includeDeleted">A value indiciting whether to include deleted users.</param>
```
Comment (bdukes) [inline DNN Platform/Library/Entities/Users/UserController.cs:654]:
```suggestion
        /// <param name="includeDeleted">A value indicating whether to include deleted users.</param>
```

File: DNN Platform/Library/Entities/Users/UserController.cs [hunk 2]
```diff
@@ -716,70 +713,68 @@ public static UserInfo GetUserById(int portalId, int userId)
...
-        /// <returns>The User as a UserInfo object.</returns>
-        /// -----------------------------------------------------------------------------
+        /// <returns>The User as a <see cref="UserInfo"/> object.</returns>
         public static UserInfo GetUserByName(int portalId, string username)
         {
             return GetCachedUser(portalId, username);
         }
 
+        /// <summary>
+        /// Gets a user by it's vanity url.
```
Comment (bdukes) [inline DNN Platform/Library/Entities/Users/UserController.cs:738]:
```suggestion
        /// Gets a user by its vanity url.
```

File: DNN Platform/Library/Entities/Users/UserController.cs [hunk 3]
```diff
@@ -716,70 +713,68 @@ public static UserInfo GetUserById(int portalId, int userId)
...
+        /// <param name="vanityUrl">The vanity url to search.</param>
+        /// <returns>A user as a <see cref="UserInfo"/> object.</returns>
         public static UserInfo GetUserByVanityUrl(int portalId, string vanityUrl)
         {
             portalId = GetEffectivePortalId(portalId);
             return MembershipProvider.Instance().GetUserByVanityUrl(portalId, vanityUrl);
         }
 
+        /// <summary>
+        /// Gets a single user by it's password reset token.
```
Comment (bdukes) [inline DNN Platform/Library/Entities/Users/UserController.cs:750]:
```suggestion
        /// Gets a single user by its password reset token.
```

**Review by bdukes — APPROVED** [review]
Thanks!

**Review by david-poindexter — APPROVED** [review]
Thanks @valadas!

**Comment by valadas** [comment]
Thanks for the typo fixes 👍

**Review by mitchelsellers — APPROVED** [review]

**Pull request merged**